panic during s3 multipart upload

<!--

Welcome :-) We understand you are having a problem with rclone; we want to help you with that!

If you've just got a question or aren't sure if you've found a bug then please use the rclone forum:

    https://forum.rclone.org/

instead of filing an issue for a quick response.

If you think you might have found a bug, please can you try to replicate it with the latest beta?

    https://beta.rclone.org/
    
If you can still replicate it with the latest beta, then please fill in the info below which makes our lives much easier.  A log with -vv will make our day :-)

Thank you

The Rclone Developers

-->

#### What is the problem you are having with rclone?
panic at this line https://github.com/rclone/rclone/blob/dfc7215bf9b7840f438d006bf7b399974f5ebf42/backend/s3/s3.go#L2271

#### What is your rclone version (output from `rclone version`)
```
drobo:~ # ./rclone version
rclone v1.51.0-DEV
- os/arch: linux/arm
- go version: go1.13.8
```

built from
dfc7215bf9b7840f438d006bf7b399974f5ebf42


#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
```
drobo:~ # uname -a
Linux drobo 3.2.96-3 #1 SMP Wed Dec 19 21:23:14 UTC 2018 armv7l GNU/Linux
```
(32bit)


####  Which cloud storage system are you using? (eg Google Drive)
aws s3


#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
`./rclone sync /[path]/ s3-private:[bucket]/[prefix]/ --progress --create-empty-src-dirs --links`


#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
took ~6 hours to occur the first time, I am running again and will provide logs if I can reproduce

```
Transferred:      126.186G / 1.435 TBytes, 9%, 6.005 MBytes/s, ETA 2d15h37m54s
Checks:             32864 / 42880, 77%
Transferred:          945 / 7351, 13%
Elapsed time:   5h58m39.1s
Checking:
 * [path]: checking
 * [path]: checking
 * [path]: checking
 * [path]: checking
 * [path]: checking
 * [path]: checking
 * [path]: checking
 * [path]: checking
Transferring:
 * [path]:  0% /61.232G, 1.739M/s, 10h0m39s
 * [path]:  6% /46.622G, 1.672M/s, 7h23m36s
 * [path]: 84% /1.403G, 1.300M/s, 2m46s
 * [path]: 29% /1.592G, 1.730M/s, 11m1s
panic: runtime error: slice bounds out of range [:7340032] with capacity 5242880

goroutine 219414 [running]:
github.com/rclone/rclone/backend/s3.(*Object).uploadMultipart.func4(0x25befb4, 0x0)
        /Users/brandon/src/rclone/backend/s3/s3.go:2271 +0x348
golang.org/x/sync/errgroup.(*Group).Go.func1(0x509fd40, 0xcd1d700)
        /Users/brandon/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:57 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go
        /Users/brandon/go/pkg/mod/golang.org/x/sync@v0.0.0-20190911185100-cd5d95a43a6e/errgroup/errgroup.go:54 +0x50
```

--- a/backend/s3/s3.go
+++ b/backend/s3/s3.go
@@ -1917,7 +1917,7 @@ func (f *Fs) getMemoryPool(size int64) *pool.Pool {
 	if !ok {
 		f.pools[size] = pool.New(
 			time.Duration(f.opt.MemoryPoolFlushTime),
-			int(f.opt.ChunkSize),
+			int(size),
 			f.opt.UploadConcurrency*fs.Config.Transfers,
 			f.opt.MemoryPoolUseMmap,
 		)

